test/messaging/00-required.e2e.spec.js

```diff
@@ -0,0 +1,49 @@
+const E2eHelpers = require('../util/e2e-helpers');
+const Timeouts = require('../util/timeouts.js');
+const MessagingHelpers = require('../util/messaging-helpers');
+const LoginHelpers = require('../util/login-helpers');
+
+if (!process.env.BUILDTYPE || process.env.BUILDTYPE === 'development') {
+  module.exports = E2eHelpers.createE2eTest(
+    (client) => {
+      MessagingHelpers.initApplicationSubmitMock();
+      LoginHelpers.logIn(client, '/healthcare/messaging', 3);
+
+      // Ensure messaging home page renders
+      client
+        .url(`${E2eHelpers.baseUrl}/healthcare/messaging`)
+        .waitForElementVisible('body', Timeouts.normal)
+        .assert.title('Send a message to your provider: Vets.gov')
+        .waitForElementVisible('#messaging-app', Timeouts.slow);
+
+      client
+        .waitForElementVisible('#messaging-nav', Timeouts.slow)
+        .waitForElementVisible('#messaging-content-header', Timeouts.slow)
+        .waitForElementVisible('#messaging-folder-controls', Timeouts.normal)
+        // expect messages to show up
+        .expect.element('.msg-table-list td:nth-of-type(1) a:nth-of-type(1)').text.to.equal('Clinician');
+
+      // Compose message view
+      client
+        .click('#messaging-folder-controls .messaging-compose-button')
```

Comment:
On desktop, the way to access it would be from the FolderNav component, while this particular button would appear only on mobile. I'm not sure about the right way to do it, but it might make sense to do mobile E2E separately?